### PR TITLE
feat: password strength checker

### LINKED ISSUE #165: Add Password Strength Checker to Enhance User Security
**Objective:**
The goal of this issue is to implement a password strength checker under the signup password field to ensure that users create strong and secure passwords, thus enhancing the overall security of our application.

**Background:**
Currently, our application lacks a mechanism to guide users in creating strong passwords. Weak passwords can pose a significant security risk. By adding a password strength checker, we can help users create strong and unique passwords, reducing the likelihood of unauthorized access to their accounts. As of now, the current system to guide user in making strong password doesn't seem that interactive and user friendly.

**Proposed Features:**
- [x] Implement a password strength meter beneath the password input field during the signup process.
- [x] Evaluate passwords against specific criteria (e.g., length, complexity, presence of uppercase, lowercase, numbers, and special characters).
- [x] Display real-time feedback to users regarding the strength of their password.
- [x] Provide guidance on how to improve weak passwords (e.g., by suggesting longer passwords or adding special characters).

Am I working on this feature improvement?
- [x] Yes `if assigned`

### PR DESCRIPTION
**Pull request by 0xAdiyat**
### **Enhance User Security with Password Strength Checker**

Incorporated a Password Strength Checker into the application to bolster user account security. The ValidatorTile widget has been updated to display password strength indicators from left to right, ensuring a more intuitive and visually appealing representation of password strength conditions. Now, the strength containers and titles are colored correctly as conditions are met, enhancing the user experience and security.

Enhance Password Strength Indicator

Modified the PasswordStrengthIndicator widget to display a color indicator that reflects the number of password strength checks met:
- If no checks are true, it shows grey color.
- If one check is true, it displays amberRed color.
- If two checks are true, it uses yellow color.
- If all three checks are true, it shows green color.

This enhancement provides more informative and visually appealing feedback on password strength.

## Screen recording

https://streamable.com/l2jysz [Link will expire in 2 days]
> As the video size exceeds 10MB, I couldn't upload it here.

enhances #165, closes #165

### CODE REVIEW

**Review by chandansgowda — CHANGES_REQUESTED** [review]
The bar should come to the center or span the whole width. Currently it looks odd since it floats to the left.

**Comment by 0xAdiyat** [comment]
> The bar should come to the center or span the whole width. Currently it looks odd since it floats to the left.

i will fix it by tomorrow

**Comment by 0xAdiyat** [comment]
image

@chandansgowda  Here is the updated screenshot as per the changes which were asked

File: lib/utils/colors.dart [hunk 1]
```diff
@@ -1,9 +1,10 @@
-
 import 'package:flutter/material.dart';
 
-class AppColor{
+class AppColor {
   static const Color yellowColor = Color(0xffFDD51F);
   static const Color greenColor = Color(0xff44AA32);
+  static const Color amberRedColor = Colors.red;
```
Comment (chandansgowda) [inline lib/utils/colors.dart:6]: Why have you named this amber red ?

**Review by chandansgowda — COMMENTED** [review]
Why have you named a colour as amber red ?

File: lib/utils/colors.dart (hunk 1 above)
Comment (0xAdiyat) [reply lib/utils/colors.dart]: > Why have you named this amber red ?

My apologies, at first, I did try using the amber-red color to check how it looks, and it seems I forgot to update it. 😅

**Review by chandansgowda — APPROVED** [review]

**Pull request merged**